tests/kafkatest/tests/core/upgrade_test.py

```diff
@@ -62,11 +62,16 @@ def perform_upgrade(self, from_kafka_version, to_message_format_version=None):
             self.kafka.start_node(node)
 
     @cluster(num_nodes=6)
+    @parametrize(from_kafka_version=str(LATEST_0_10_2), to_message_format_version=str(LATEST_0_9), compression_types=["none"])
```

Comment:
Discussed offline. This one looks a little funny. We shouldn't be able to downgrade the message format.